plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestSharedHiveMetastore.java

```diff
@@ -0,0 +1,194 @@
+        assertQuerySucceeds("DROP TABLE IF EXISTS iceberg." + schema + ".nation");
+        assertQuerySucceeds("DROP SCHEMA IF EXISTS hive." + schema);
+    }
+
+    @Test
+    public void testReadInformationSchema()
+    {
+        assertThat(query("SELECT table_schema FROM hive.information_schema.tables WHERE table_name = 'region'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + schema + "'");
+        assertThat(query("SELECT table_schema FROM iceberg.information_schema.tables WHERE table_name = 'nation'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + schema + "'");
+        assertThat(query("SELECT table_schema FROM hive_with_redirections.information_schema.tables WHERE table_name = 'region'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + schema + "'");
+        assertThat(query("SELECT table_schema FROM hive_with_redirections.information_schema.tables WHERE table_name = 'nation'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + schema + "'");
+        assertThat(query("SELECT table_schema FROM iceberg_with_redirections.information_schema.tables WHERE table_name = 'region'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + schema + "'");
+
+        assertQuery("SELECT table_name, column_name from hive.information_schema.columns WHERE table_schema = '" + schema + "'",
+                "VALUES ('region', 'regionkey'), ('region', 'name'), ('region', 'comment')");
+        assertQuery("SELECT table_name, column_name from iceberg.information_schema.columns WHERE table_schema = '" + schema + "'",
+                "VALUES ('nation', 'nationkey'), ('nation', 'name'), ('nation', 'regionkey'), ('nation', 'comment')");
+        assertQuery("SELECT table_name, column_name from hive_with_redirections.information_schema.columns WHERE table_schema = '" + schema + "'",
+                "VALUES" +
+                        "('region', 'regionkey'), ('region', 'name'), ('region', 'comment'), " +
+                        "('nation', 'nationkey'), ('nation', 'name'), ('nation', 'regionkey'), ('nation', 'comment')");
+        assertQuery("SELECT table_name, column_name from iceberg_with_redirections.information_schema.columns WHERE table_schema = '" + schema + "'",
+                "VALUES" +
+                        "('nation', 'nationkey'), ('nation', 'name'), ('nation', 'regionkey'), ('nation', 'comment')");
+    }
+
+    @Test
+    public void testShowTables()
+    {
+        assertQuery("SHOW TABLES FROM iceberg." + schema, "VALUES 'nation'");
+        assertQuery("SHOW TABLES FROM hive." + schema, "VALUES 'region', 'nation'");
+        assertQuery("SHOW TABLES FROM hive_with_redirections." + schema, "VALUES 'region', 'nation'");
+        assertQuery("SHOW TABLES FROM iceberg_with_redirections." + schema, "VALUES 'nation'");
```

Comment:
With the Hive metastore , the catalog implementation is filtering out the non-Iceberg tables, so we don't see here the `region` table as we see it in case of dealing with `Glue`.

cc: @findepi @alexjo2144

Reply:
ACK

Reply:
Relates to https://github.com/trinodb/trino/pull/11617 right?